### DISCUSSION TITLE
More power with the soyo 😬

### DISCUSSION THREAD

**Discussion by kev300** [syssi/esphome-soyosource-gtn-virtual-meter#145] [Show and tell]
Hi, not sure if this is only possible with my newer purple soyosource inverter...

I was not aware that the limiter can also be used in PV mode and the required settings part in the docs of this project says that the Battery limit mode is required.

But at the moment I'm running on battery in PV mode with the limiter, which raises the 900W limit to 1200W (on the gtn-1200).
I think thats realy cool and worth noting... 🚀 

You need a higher battery voltage for that of course. I'm using a DC-DC converter to boost to 70V so the PV mode has enough voltage to reach max power and stay below 20A, which makes the soyo a bit more efficient. The minimum voltage should be around 55V for the PV mode.

As a side effect: I can now also drain my battery to 40V without throttling of the inverter. Previously the soyo started power throttling below 48V and shutdown at 43V.

**Comment by Chickenbreast0** [comment]
Have you calculated how much power is lost through the DC/DC converter and whether it is still worthwhile in terms of efficiency?
Eventhough is it nice to know ;)

**Comment by kev300** [comment]
Not yet, but according to the spec and some tests on YouTube, the dc boost converter should only lose 4-7%.
I know that the soyo looses a lot under 46V, so if the soyo gets more efficient because of the higher voltage, the total loss might be not that much. But I will measure it myself and report 😁

**Comment by kev300** [comment]
Only tested untill 500W with measurements until now.

<= 300W efficiency loss from 93% -> 90%
400W 92,6% -> 88,9%
500W 92,4% -> 86%

because of the weather my battery is constantly empty, so did not test higher wattages, yet...

**Reply by Chickenbreast0** [reply]
Now that is measured with the DC to DC converter or without?
Somewhere I had read that the inverters have their sweet spot at about 500W at the highest efficiency.

**Reply by kev300** [reply]
both. So the higher number is without and the lower is with the DC/DC converter boosted to 60V.

But obviously not measured the interesting load settings. :D

**Reply by Chickenbreast0** [reply]
Ah sorry...then I had not read it correctly. But good that you can already see the power loss. Interesting that you did it, but I'm not sure if the efficiency ratio is so more worth it.

**Comment by kev300** [comment]
In general, the additional efficiency loss of the boost converter is around 3%-5,5%
Here are my measurements:

| Demand setting  | Efficiency with DC/DC Booster | only Soyosource Efficiency |  
| ------------- | ------------- | ------------ |
| 100W | 89,2% | 89,5% |
| 300W | 91,3%  | 93,7% |
| 400W | 91,2% | 94,1% |
| 500W | 91,2% | 93,5% |
| 700W | -- | 92,9% |
| 800W | 89,9% | 92,3% |
| 900W | 88% | 91,4% |
| 1100W | 87,7% | -- |
| 1200W | 87,2% | -- |

So, yes, there is a loss, but if you want more power and have excess solar, it can be worth it. 😬

**Comment by ericgrand** [comment]
Hi! 
Is the DC-DC converter act as a current limiter? If the battery were 72V, would the soyo try to draw more amps?
I'm questionning myself in using a 55-90V version with 6 12V LFP batteries in serie...

**Reply by kev300** [reply]
More voltage -> less amps.
My soyo is self limited to 25A at low voltages, but beginning with 52v it's power limited. If i give it 80V, it only draws 16A max.
So, no it won't go above it's rated power.

The dc dc converter was not limiting the current in my case.

**Reply by ericgrand** [reply]
Many thanks, so a version 55-90V used with battery up to 86V should be fine.
Is max power limit parameter still available in PV mode?

**Reply by kev300** [reply]
I think so, but I only limited it via the RS 485 interface. So not sure.

**Reply by syssi** [reply]
I think so too. Even if the RS485 limiter is used the max.power limit setting defines the upper limit. In other words if the max power limit is set to 200W and a power demand of 300W is requested using RS485 the inverter will deliver 200W. I hope this applies to the PV mode + Limiter enabled too.

**Reply by kev300** [reply]
Mh, but you can only set the max limit to 900W, right? But the PV mode is up to 1200W, so I guess it's not used in PV mode.

**Reply by syssi** [reply]
Good point!

**Reply by ericgrand** [reply]
Ok, the important point for me is that the inverter will not try to draw as much amps as the batteries are able to deliver...
As the inverter will always be driven by the limiter, it is fine for me.

**Comment by sorz2122** [comment]
Which DC-DC converter did you use?

**Comment by kev300** [comment]
Google: 1800W 40A DC-DC Boost Converter Step Up Power Supply Module
For about 22€

**Reply by sorz2122** [reply]
TY
